src/lib/core/CHIPTLVReader.cpp

```diff
@@ -781,6 +782,13 @@ CHIP_ERROR TLVReader::EnsureData(CHIP_ERROR noDataErr)
         if (bufLen == 0)
             return noDataErr;
 
+        backingStore = mBackingStore->GetNextBackingStore();
```

Comment:
See, this is the part I don't understand.  Why do we need the change here, and to `TLVBackingStore`?  Why can't we just have a `TLVBackingStore` implementation whose `GetNextBuffer` does the right thing and keeps the fact that it's actually a combination of several backing stores as an implementation detail?  It could be `CircularEventBuffer` itself, or it could be a backing store that knows about how `CircularEventBuffer` is set up and acts accordingly, or it could be a generic "I store a list of other backing stores and fail over from one to the other as needed", depending on how we want to set it up.....  But it seems like the `TLVBackingStore` abstraction is simple enough and flexible enough to represent the thing we want here without adding special-casing to the reader.

Reply:
agree, thanks, will update this in chip eventing pull request, and keep only GetRemainingFreeLength in this pr